<!DOCTYPE html>



  


<html class="theme-next muse use-motion" lang="en">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>
<meta name="theme-color" content="#222">









<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />
















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />







<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=5.1.4" rel="stylesheet" type="text/css" />


  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png?v=5.1.4">


  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png?v=5.1.4">


  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png?v=5.1.4">


  <link rel="mask-icon" href="/images/logo.svg?v=5.1.4" color="#222">





  <meta name="keywords" content="Tools,Linux," />










<meta name="description" content="说明该流程涵盖了宏基因组分析的大部分代码">
<meta name="keywords" content="Tools,Linux">
<meta property="og:type" content="article">
<meta property="og:title" content="宏基因组分析流程">
<meta property="og:url" content="http://yoursite.com/2019/10/30/宏基因组分析流程/index.html">
<meta property="og:site_name" content="H&amp;J Home">
<meta property="og:description" content="说明该流程涵盖了宏基因组分析的大部分代码">
<meta property="og:locale" content="en">
<meta property="og:updated_time" content="2019-10-30T14:00:54.248Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="宏基因组分析流程">
<meta name="twitter:description" content="说明该流程涵盖了宏基因组分析的大部分代码">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Muse',
    version: '5.1.4',
    sidebar: {"position":"left","display":"post","offset":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: {"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},
    duoshuo: {
      userId: '0',
      author: 'Author'
    },
    algolia: {
      applicationID: 'AURFS67F7M',
      apiKey: '',
      indexName: 'hao',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://yoursite.com/2019/10/30/宏基因组分析流程/"/>





  <title>宏基因组分析流程 | H&J Home</title>
  








</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="en">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="/"  class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">H&J Home</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            Home
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br />
            
            About
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br />
            
            Tags
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br />
            
            Categories
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br />
            
            Archives
          </a>
        </li>
      

      
    </ul>
  

  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://yoursite.com/2019/10/30/宏基因组分析流程/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="Hao Zhou">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="http://p39keqca4.bkt.clouddn.com/blog/180128/Ibm7g4cDjf.jpg?imageslim">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="H&J Home">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">宏基因组分析流程</h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">Posted on</span>
              
              <time title="Post created" itemprop="dateCreated datePublished" datetime="2019-10-30T22:00:54+08:00">
                2019-10-30
              </time>
            

            

            
          </span>

          
            <span class="post-category" >
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">In</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/metagenomic-data-analysis/" itemprop="url" rel="index">
                    <span itemprop="name">metagenomic data analysis</span>
                  </a>
                </span>

                
                
              
            </span>
          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <h2 id="说明"><a href="#说明" class="headerlink" title="说明"></a>说明</h2><p>该流程涵盖了宏基因组分析的大部分代码<br><a id="more"></a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">#!/usr/bin/env bash</span><br></pre></td></tr></table></figure></p>
<h3 id="原始数据质量检测"><a href="#原始数据质量检测" class="headerlink" title="原始数据质量检测"></a>原始数据质量检测</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">fc rowdata</span><br><span class="line">project=/home/menghe/hj/project/hlow_metagenome</span><br><span class="line">ls $&#123;project&#125;/0-rawdata/highsample/????.R?.fastq.gz |while read id; do fastqc $id;done</span><br></pre></td></tr></table></figure>
<h3 id="序列过滤，去掉低质量和宿主序列"><a href="#序列过滤，去掉低质量和宿主序列" class="headerlink" title="序列过滤，去掉低质量和宿主序列"></a>序列过滤，去掉低质量和宿主序列</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ls $&#123;project&#125;/0-rawdata/highsample/????.R1.fastq.gz |while read id; do \</span><br><span class="line">	kneaddata --input $&#123;id%%.*&#125;.R1.fastq.gz --input $&#123;id%%.*&#125;.R2.fastq.gz --threads 8 --processes 8  --reference-db $&#123;project&#125;/reference/gg.db --output $&#123;id%%.*&#125; --max-memory 5000m --run-fastqc-end --trimmomatic-options=&quot;ILLUMINACLIP:/home/menghe/anaconda3/envs/metagenome/share/trimmomatic-0.39-1/adapters/NexteraPE-PE.fa:2:40:15&quot; --trimmomatic-options=&quot;LEADING:2&quot; --trimmomatic-options=&quot;TRAILING:2&quot; --trimmomatic-options=&quot;SLIDINGWINDOW:4:2&quot; --trimmomatic-options=&quot;MINLEN:25&quot; --bowtie2-options=&quot;-1 $&#123;id%%.*&#125;.R1.fastq.gz&quot; --bowtie2-options=&quot;-2 $&#123;id%%.*&#125;.R2.fastq.gz&quot; \</span><br><span class="line">	;done</span><br></pre></td></tr></table></figure>
<h3 id="质量检测-fastqc"><a href="#质量检测-fastqc" class="headerlink" title="质量检测 fastqc"></a>质量检测 fastqc</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ls $&#123;project&#125;/0-rawdata/highsample/????.R1.fastq.gz |while read id; do \</span><br><span class="line">	cd $&#123;id%%.*&#125; &amp;&amp; rm ????.R1_kneaddata.trimmed.1.fastq ????.R1_kneaddata.trimmed.2.fastq &amp;&amp; rename &apos;s/R1_//&apos; ????.R1_* \</span><br><span class="line">	;done</span><br></pre></td></tr></table></figure>
<h3 id="统计reads个数"><a href="#统计reads个数" class="headerlink" title="统计reads个数"></a>统计reads个数</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">ls *.fastq.gz &gt;high.clean.list</span><br><span class="line">for i in $(cat high.clean.list); do printf $i &quot;\t&quot; &gt;&gt;fq.high.reads_num;readfq $i &gt;&gt;fq.high.reads_num; done</span><br></pre></td></tr></table></figure>
<h3 id="最简单的菌群注释-metaphlan2"><a href="#最简单的菌群注释-metaphlan2" class="headerlink" title="最简单的菌群注释 metaphlan2"></a>最简单的菌群注释 metaphlan2</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ls $&#123;project&#125;/1-cleandata/????.kneaddata_paired_1.fastq |while read id; do \</span><br><span class="line">	metaphlan2.py --input_type fastq --bowtie2out $&#123;id%%.*&#125;.bowtie2.bz2 --nproc 12  &lt;(cat $&#123;id%%.*&#125;.kneaddata_paired_1.fastq $&#123;id%%.*&#125;.kneaddata_paired_2.fastq ) &gt; $&#123;id%%.*&#125;.metaphlan.txt \</span><br><span class="line">	;done</span><br></pre></td></tr></table></figure>
<h3 id="合并各样本双端序列为一个文件"><a href="#合并各样本双端序列为一个文件" class="headerlink" title="合并各样本双端序列为一个文件"></a>合并各样本双端序列为一个文件</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ls $&#123;project&#125;/1-cleandata/????.kneaddata_paired_1.fastq |while read id; do \</span><br><span class="line">	interleave-reads.py $&#123;id%%.*&#125;.kneaddata_paired_1.fastq $&#123;id%%.*&#125;.kneaddata_paired_2.fastq --gzip -o $&#123;id%%.*&#125;.kneaddata.com.fastq.gz \</span><br><span class="line">	;done</span><br></pre></td></tr></table></figure>
<h3 id="删除-kmer-（选做）"><a href="#删除-kmer-（选做）" class="headerlink" title="删除 kmer （选做）"></a>删除 kmer （选做）</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">abundance-dist-single.py -M 2e9 -k 21 $&#123;rawdata&#125;/3983_R1.fastq.gz $&#123;rawdata&#125;/3983_R1.fastq.gz.dist</span><br><span class="line">wait</span><br><span class="line">abundance-dist-single.py -M 2e9 -k 21 $&#123;project&#125;/3983_R1.qc.fastq.gz $&#123;project&#125;/3983_R1.qc.fastq.gz.dist</span><br><span class="line">wait</span><br><span class="line">trim-low-abund.py -V -M 10e9 -C 3 -Z 10 --gzip -o $&#123;project&#125;/1-cleandata/3983_R1.qc.pairkmer.fastq $&#123;project&#125;/1-cleandata/3983_R1.qc.paired.fastq.gz</span><br><span class="line">wait</span><br><span class="line">unique-kmers.py $&#123;project&#125;/1-cleandata/3983_R1.qc.kmer.fastq</span><br></pre></td></tr></table></figure>
<h3 id="将reads组装成contig"><a href="#将reads组装成contig" class="headerlink" title="将reads组装成contig"></a>将reads组装成contig</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">for i in &#123;4171,4199,4203,4259,4277,4278,4282,4287,4291,4334&#125;; do</span><br><span class="line">	megahit -1 $&#123;project&#125;/1-cleandata/$&#123;i&#125;.kneaddata_paired_1.fastq.gz -2 $&#123;project&#125;/1-cleandata/$&#123;i&#125;.kneaddata_paired_2.fastq.gz -t 15 -o $&#123;project&#125;/2-Assembly/$&#123;i&#125;.kneaddata.megahit ;done</span><br></pre></td></tr></table></figure>
<h3 id="检测组装质量"><a href="#检测组装质量" class="headerlink" title="检测组装质量"></a>检测组装质量</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">cd ./2-assembly/</span><br><span class="line">for i in $(cat id.txt); do</span><br><span class="line">	cd $&#123;i&#125;.kneaddata.megahit &amp;&amp; quast final.contigs.fa -o $&#123;i&#125;.quast.report &amp;&amp; cd ..;done</span><br></pre></td></tr></table></figure>
<h3 id="用contig-预测蛋白"><a href="#用contig-预测蛋白" class="headerlink" title="用contig 预测蛋白"></a>用contig 预测蛋白</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">for i in $(cat /ifs1/Grp3/stu142/metagenome/3-assembly/id.2.txt); do</span><br><span class="line">prokka /ifs1/Grp3/stu142/metagenome/3-assembly/$&#123;i&#125;.final.contigs.fa --outdir ./3-assembly/$&#123;i&#125;_annotation  --prefix $&#123;i&#125;G --metagenome --kingdom Bacteria ;done</span><br></pre></td></tr></table></figure>
<h3 id="合并所有样本预测蛋白核酸序列，构建非冗余基因集"><a href="#合并所有样本预测蛋白核酸序列，构建非冗余基因集" class="headerlink" title="合并所有样本预测蛋白核酸序列，构建非冗余基因集"></a>合并所有样本预测蛋白核酸序列，构建非冗余基因集</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">cat ./3-assembly/$&#123;i&#125;_annotation/????G.ffn &gt;all.fa</span><br><span class="line">cd-hit-est -i all.fa -o all.ffn.merged -c 0.95 -n 10 -d 0 -M 16000 - T 8</span><br></pre></td></tr></table></figure>
<h3 id="非冗余基因集index构建"><a href="#非冗余基因集index构建" class="headerlink" title="非冗余基因集index构建"></a>非冗余基因集index构建</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">salmon index -t all.ffn.merged -i all.ffn.merged_index --type quasi -k 31</span><br></pre></td></tr></table></figure>
<h3 id="map-to-the-reference"><a href="#map-to-the-reference" class="headerlink" title="map to the reference"></a>map to the reference</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">for i in $(cat $&#123;project&#125;/1-cleandata/id.txt); do </span><br><span class="line">salmon quant -i $&#123;project&#125;/4-predict/all.ffn.merged_index -p 10 --libType IU -1 $&#123;project&#125;/1-cleandata/$&#123;i&#125;.kneaddata_paired_1.fastq.gz -2 $&#123;project&#125;/1-cleandata/$&#123;i&#125;.kneaddata_paired_2.fastq.gz -o $&#123;project&#125;/4-predict/$&#123;i&#125;.quant ;done</span><br></pre></td></tr></table></figure>
<h3 id="统计每个样本中的基因丰度"><a href="#统计每个样本中的基因丰度" class="headerlink" title="统计每个样本中的基因丰度"></a>统计每个样本中的基因丰度</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">curl -L -O https://raw.githubusercontent.com/ngs-docs/2016-metagenomics-sio/master/gather-counts.py</span><br><span class="line">chmod +x gather-counts.py</span><br><span class="line">./gather-counts.py</span><br></pre></td></tr></table></figure>
<h3 id="合并成丰度矩阵"><a href="#合并成丰度矩阵" class="headerlink" title="合并成丰度矩阵"></a>合并成丰度矩阵</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">for file in *counts</span><br><span class="line">do</span><br><span class="line">  # 提取样品名</span><br><span class="line">  name=$&#123;file%%.*&#125;</span><br><span class="line">  # 将每个文件中的count列改为样品列</span><br><span class="line">  sed -e &quot;s/count/$name/g&quot; $file &gt; tmp</span><br><span class="line">  mv tmp $file</span><br><span class="line">done</span><br><span class="line">  # 合并所有样品</span><br><span class="line">paste *counts |cut -f 1,2,4 &gt; Combined-counts.tab</span><br></pre></td></tr></table></figure>
<h3 id="利用每个样本中有表达量的gene做功能注释-（2号服务器）"><a href="#利用每个样本中有表达量的gene做功能注释-（2号服务器）" class="headerlink" title="利用每个样本中有表达量的gene做功能注释 （2号服务器）"></a>利用每个样本中有表达量的gene做功能注释 （2号服务器）</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"> ~/hj/project/hlow_metagenome/4-predict</span><br><span class="line">python gather-counts.py #从quant结果中提取出基因名和read counts</span><br><span class="line">ls *quant.counts |while read id ; do  awk &apos;$2!=0 &amp;&amp; $2!=&quot;count&quot; &#123;print $1&#125;&apos; $&#123;id&#125; &gt;$&#123;id&#125;.no0.id ;done   #提取出表达gene 的ID</span><br><span class="line">for i in $(cat id.txt); do python extrac_seq.py $&#123;i&#125;.quant.counts.no0.id all.ffn.merged  $&#123;i&#125;.extract.fq ;done #利用ID 提取表达基因序列</span><br></pre></td></tr></table></figure>
<h3 id="利用提取的序列进行-宏基因组kegg等功能分析"><a href="#利用提取的序列进行-宏基因组kegg等功能分析" class="headerlink" title="利用提取的序列进行 宏基因组kegg等功能分析"></a>利用提取的序列进行 宏基因组kegg等功能分析</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">ls 5-function/*.faa|while read id; do ./eggnog-mapper/emapper.py -i $&#123;id&#125;  --output $&#123;id&#125;.maNOG  -m diamond --cpu 8 --usemem ;done</span><br><span class="line">#https://github.com/eggnogdb/eggnog-mapper/wiki/eggNOG-mapper-v2</span><br></pre></td></tr></table></figure>
      
    </div>
    
    
    

    

    

    

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/Tools/" rel="tag"># Tools</a>
          
            <a href="/tags/Linux/" rel="tag"># Linux</a>
          
        </div>
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/2019/10/30/microbiome GWAS/" rel="next" title="microbiome GWAS 分析流程">
                <i class="fa fa-chevron-left"></i> microbiome GWAS 分析流程
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          

  



        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap">
            Table of Contents
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview-wrap">
            Overview
          </li>
        </ul>
      

      <section class="site-overview-wrap sidebar-panel">
        <div class="site-overview">
          <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
            
              <img class="site-author-image" itemprop="image"
                src="http://p39keqca4.bkt.clouddn.com/blog/180128/Ibm7g4cDjf.jpg?imageslim"
                alt="Hao Zhou" />
            
              <p class="site-author-name" itemprop="name">Hao Zhou</p>
              <p class="site-description motion-element" itemprop="description"></p>
          </div>

          <nav class="site-state motion-element">

            
              <div class="site-state-item site-state-posts">
              
                <a href="/archives/">
              
                  <span class="site-state-item-count">21</span>
                  <span class="site-state-item-name">posts</span>
                </a>
              </div>
            

            
              
              
              <div class="site-state-item site-state-categories">
                <a href="/categories/index.html">
                  <span class="site-state-item-count">6</span>
                  <span class="site-state-item-name">categories</span>
                </a>
              </div>
            

            
              
              
              <div class="site-state-item site-state-tags">
                <a href="/tags/index.html">
                  <span class="site-state-item-count">6</span>
                  <span class="site-state-item-name">tags</span>
                </a>
              </div>
            

          </nav>

          

          

          
          

          
          

          

        </div>
      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#说明"><span class="nav-number">1.</span> <span class="nav-text">说明</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#原始数据质量检测"><span class="nav-number">1.1.</span> <span class="nav-text">原始数据质量检测</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#序列过滤，去掉低质量和宿主序列"><span class="nav-number">1.2.</span> <span class="nav-text">序列过滤，去掉低质量和宿主序列</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#质量检测-fastqc"><span class="nav-number">1.3.</span> <span class="nav-text">质量检测 fastqc</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#统计reads个数"><span class="nav-number">1.4.</span> <span class="nav-text">统计reads个数</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#最简单的菌群注释-metaphlan2"><span class="nav-number">1.5.</span> <span class="nav-text">最简单的菌群注释 metaphlan2</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#合并各样本双端序列为一个文件"><span class="nav-number">1.6.</span> <span class="nav-text">合并各样本双端序列为一个文件</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#删除-kmer-（选做）"><span class="nav-number">1.7.</span> <span class="nav-text">删除 kmer （选做）</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#将reads组装成contig"><span class="nav-number">1.8.</span> <span class="nav-text">将reads组装成contig</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#检测组装质量"><span class="nav-number">1.9.</span> <span class="nav-text">检测组装质量</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#用contig-预测蛋白"><span class="nav-number">1.10.</span> <span class="nav-text">用contig 预测蛋白</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#合并所有样本预测蛋白核酸序列，构建非冗余基因集"><span class="nav-number">1.11.</span> <span class="nav-text">合并所有样本预测蛋白核酸序列，构建非冗余基因集</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#非冗余基因集index构建"><span class="nav-number">1.12.</span> <span class="nav-text">非冗余基因集index构建</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#map-to-the-reference"><span class="nav-number">1.13.</span> <span class="nav-text">map to the reference</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#统计每个样本中的基因丰度"><span class="nav-number">1.14.</span> <span class="nav-text">统计每个样本中的基因丰度</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#合并成丰度矩阵"><span class="nav-number">1.15.</span> <span class="nav-text">合并成丰度矩阵</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#利用每个样本中有表达量的gene做功能注释-（2号服务器）"><span class="nav-number">1.16.</span> <span class="nav-text">利用每个样本中有表达量的gene做功能注释 （2号服务器）</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#利用提取的序列进行-宏基因组kegg等功能分析"><span class="nav-number">1.17.</span> <span class="nav-text">利用提取的序列进行 宏基因组kegg等功能分析</span></a></li></ol></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">&copy; <span itemprop="copyrightYear">2019</span>
  <span class="with-love">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Hao Zhou</span>

  
</div>


  <div class="powered-by">Powered by <a class="theme-link" target="_blank" href="https://hexo.io">Hexo</a></div>



  <span class="post-meta-divider">|</span>



  <div class="theme-info">Theme &mdash; <a class="theme-link" target="_blank" href="https://github.com/iissnan/hexo-theme-next">NexT.Muse</a> v5.1.4</div>




        







        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  












  
  
    <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>
  

  
  
    <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>
  

  
  
    <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>
  


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.4"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.4"></script>



  
  

  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.4"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.4"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.4"></script>



  


  




	





  





  












  





  

  

  

  
  

  

  

  

</body>
</html>
